File: DataStructures/Array.js
// Criação de um array vazio
let arrayVazio = [];

// Criação de um array com elementos
let arrayElementos = [1, 2, 3, 4, 5];

// Criação de um array com diferentes tipos de dados
let arrayTiposDiferentes = [1, "dois", true, { nome: "João" }];

// Criação de um array multidimensional
let arrayMultidimensional = [[1, 2], [3, 4], [5, 6]];

// Acessando elementos do array
console.log(arrayElementos[0]); // Saída: 1
console.log(arrayTiposDiferentes[2]); // Saída: true

// Modificando elementos do array
arrayElementos[0] = 10;
console.log(arrayElementos); // Saída: [10, 2, 3, 4, 5]

// Adicionando elementos ao array
arrayElementos.push(6);
console.log(arrayElementos); // Saída: [10, 2, 3, 4, 5, 6]

// Removendo elementos do array
arrayElementos.pop();
console.log(arrayElementos); // Saída: [10, 2, 3, 4, 5]

// Obtendo o tamanho do array
console.log(arrayElementos.length); // Saída: 5

File: DataStructures/Array.lua
local meuArray = {1, 2, 3, 4, 5}
-- Criando um array vazio
local meuArray = {}

-- Adicionando elementos ao array
meuArray[1] = 1
meuArray[2] = 2
meuArray[3] = 3
meuArray[4] = 4
meuArray[5] = 5

-- Acessando elementos do array
print(meuArray[1]) -- Saída: 1
print(meuArray[3]) -- Saída: 3

-- Alterando elementos do array
meuArray[2] = 10
print(meuArray[2]) -- Saída: 10

-- Obtendo o tamanho do array
local tamanho = #meuArray
print(tamanho) -- Saída: 5
